### PR TITLE
XW-2823 use json to accept curated content data

### PR DESCRIPTION
**Pull request by idradm**
https://wikia-inc.atlassian.net/browse/XW-2823
https://github.com/Wikia/mercury/pull/3091

### CODE REVIEW

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 1]
```diff
@@ -361,11 +371,18 @@ public function getData() {
 
 				$data = array_map( function ( $section ) {
 					$section[ 'node_type' ] = 'section';
+					$section['items'] = array_filter( $section['items'], function ( $item ) {
+						return Title::newFromText( $item['title'] )->isKnown();
```
Comment (kvas-damian) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: Please make sure `isKnown()` returns true for categories without article.

**Comment by kvas-damian** [comment]
Please file a ticket to not display broken category links on CMP when category got deleted.

**Comment by idradm** [comment]
https://wikia-inc.atlassian.net/browse/XW-2855

**Review by kvas-damian — APPROVED** [review]

File: extensions/wikia/CuratedContent/CuratedContentController.class.php [hunk 2]
```diff
@@ -354,18 +366,32 @@ public function getData() {
...
 				$optional = $this->communityDataService->getOptional();
 				if ( !empty( $optional ) ) {
 					$curated[] = $optional;
 				}
 
-				$data = array_map( function ( $section ) {
-					$section[ 'node_type' ] = 'section';
-					$section[ 'items' ] = $this->extendItemsWithImages( $section[ 'items' ] );
-					$section[ 'items' ] = $this->extendItemsWithType( $section[ 'items' ] );
+				$data = array_map( function ( $section ) use ( $validator ) {
```
Comment (hakubo) [inline extensions/wikia/CuratedContent/CuratedContentController.class.php]: what do you think about extracting these two loops to separate method for clarity ?

**Review by hakubo — APPROVED** [review]

**Pull request merged**